pkg/reconciler/route/config/routing.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	RoutingConfigName = "config-routing"
+
+	IngressServiceNameKey    = "ingress.servicename"
+	IngressNamespaceKey      = "ingress.namespace"
+	KnativeIngressGatewayKey = "knative.ingress.gateway"
+
+	DefaultIngressServiceName    = "istio-ingressgateway"
+	DefaultIngressNamespace      = "istio-system"
+	DefaultKnativeIngressGateway = "knative-ingress-gateway"
+)
+
+// RoutingConfig contains the networking configuration defined in the
+// network config map.
+type RoutingConfig struct {
+	// Name of the Istio Ingress svc
+	IngressServiceName string
+
+	// K8s namespace to search for Ingresses
+	IngressNamespace string
+
+	// Name of ingress gateway in knative-serving namespace
+	KnativeIngressGateway string
+
+	// Name of gateway set as route destination host in the kf virtualservice
+	// (usually back to the ingress gateway or cluster local gateway)
+	GatewayHost string
```

Comment:
nit: since this is dynamically generated, it should probably be a function

Reply:
done